server/server.go

```diff
@@ -820,7 +820,14 @@ func (s *Server) verifyAccountClaims(claimJWT string) (*jwt.AccountClaims, strin
 // Lock should be held upon entry.
 func (s *Server) fetchAccount(name string) (*Account, error) {
 	if accClaims, claimJWT, err := s.fetchAccountClaims(name); accClaims != nil {
-		acc := s.buildInternalAccount(accClaims)
+		// We have released the lock during the low level fetch.
+		// Now that we are back under lock, check again if account
+		// is in the map or not. If it is, simply return it.
+		acc := s.accounts[name]
+		if acc != nil {
+			return acc, nil
```

Comment:
What about the case where we might have a newer claim here and may want to apply it?

Reply:
Not sure how to do that.. feel free to update PR or tell me what to do here.

Reply:
ok will take a look. Working on another PR atm.